solidity/contracts/libs/isms/MerkleRootMultisigIsmMetadata.sol

```diff
@@ -3,27 +3,35 @@ pragma solidity >=0.8.0;
 
 /**
  * Format of metadata:
- * [   0:  32] Merkle root
- * [  32:  36] Root index
- * [  36:  68] Origin mailbox address
+ * [   0:  32] Origin mailbox address
+ * [  32:  36] Signed root index
+ * [  36:  68] Signed message ID
  * [  68:1092] Merkle proof
- * [1092:????] Validator signatures, 65 bytes each, length == Threshold
+ * [1092:????] Validator signatures (length := threshold)
  */
-library MultisigIsmMetadata {
-    uint256 private constant MERKLE_ROOT_OFFSET = 0;
-    uint256 private constant MERKLE_INDEX_OFFSET = 32;
-    uint256 private constant ORIGIN_MAILBOX_OFFSET = 36;
-    uint256 private constant MERKLE_PROOF_OFFSET = 68;
-    uint256 private constant SIGNATURES_OFFSET = 1092;
-    uint256 private constant SIGNATURE_LENGTH = 65;
+library MerkleRootMultisigIsmMetadata {
+    uint8 private constant ORIGIN_MAILBOX_OFFSET = 0;
+    uint8 private constant MERKLE_INDEX_OFFSET = 32;
+    uint8 private constant SIGNED_MESSAGE_ID_OFFSET = 36;
+    uint8 private constant MERKLE_PROOF_OFFSET = 68;
+    uint16 private constant MERKLE_PROOF_LENGTH = 32 * 32;
+    uint16 private constant SIGNATURES_OFFSET = 1092;
```

Comment:
MERKLE_PROOF_OFFSET + MERKLE_PROOF_LENGTH?

Reply:
we could do this but it feels like a departure from the current pattern?